### PR TITLE
Move to new ilasm version

### LINKED ISSUE #37582: NETSDK1056: Project is targeting runtime 'win-x64' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.	ILTools
**Version used**: VisualStudio.16.Preview/16.3.0-pre.1.0+29123.89
C# Tools 3.3.0-beta1-19360-03+7d47461d2ddd32defbc51c2cf0e75a3fef16c3d7

I have been getting this error occasionally on incremental builds of Roslyn.sln. It doesn't seem to stop me from running my tests but it is a bit confusing.

`NETSDK1056: Project is targeting runtime 'win-x64' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.	ILTools	F:\src\roslyn\src\Tools\ILTools\ILTools.proj	1`

### PR DESCRIPTION
**Pull request by jaredpar**
The latest version of ilasm produced by coreclr is now a fully
independent executable; it no longer depends on having a full runtime
laid down next to it.

This means we can vastly simplify how it is deployde in our
infrastructure. The package can now be included and have ilasm manually
copied out. This is similar to the approach that we take for our
diasymreader native dependencies.

closes #37582
related #25930, #25144

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure, @nguerrera, @tmat for review

File: src/Test/Utilities/Portable/Metadata/IlasmUtilities.cs [hunk 1]
```diff
@@ -21,36 +21,27 @@ public static DisposableFile CreateTempAssembly(string declarations, bool prepen
...
-                    directory = Path.GetDirectoryName(directory);
-                }
-
-                if (directory == null)
-                {
-                    throw new NotSupportedException("Unable to find CoreCLR ilasm tool. Has the Microsoft.NETCore.ILAsm package been published to /artifacts/tools?");
-                }
-
-                return path;
+                throw new Exception("Runtime platform not supported for testing");
```
Comment (nguerrera) [inline src/Test/Utilities/Portable/Metadata/IlasmUtilities.cs]:
```suggestion
                throw new PlatformNotSupportedException("Runtime platform not supported for testing");
```

**Comment by AlekseyTs** [comment]
>     <MicrosoftNetCoreILAsmVersion>3.0.0-preview6-27721-71</MicrosoftNetCoreILAsmVersion>

Do we still need this property and the one below?

---
Refers to: eng/Versions.props:76 in bc0d55c. [](commit_id = bc0d55c95df7f152c2cba0c84f218da178bb6778, deletion_comment = False)

**Comment by jaredpar** [comment]
>     <MicrosoftNetCoreILAsmVersion>3.0.0-preview6-27721-71</MicrosoftNetCoreILAsmVersion>

Good catch. Don't think that is used anymore. 

---
In reply to: [520556648](https://github.com/dotnet/roslyn/pull/37878#issuecomment-520556648) [](ancestors = 520556648)

---
Refers to: eng/Versions.props:76 in bc0d55c. [](commit_id = bc0d55c95df7f152c2cba0c84f218da178bb6778, deletion_comment = False)

**Pull request merged**